prow/spyglass/podlogartifact.go

```diff
@@ -152,6 +158,9 @@ func (a *PodLogArtifact) ReadAtMost(n int64) ([]byte, error) {
 
 // ReadTail reads the last n bytes of the pod log
 func (a *PodLogArtifact) ReadTail(n int64) ([]byte, error) {
+	if n > a.sizeLimit {
```

Comment:
if n > a.sizeLimit but there are less than `a.sizeLimit` total logs, should we still return error?

Reply:
That is a good question and I waffled on the implementation. I finally decided to just keep the implementation simple. Thoughts? Reasons to change it?

If we really wanted to be an awesome API, we would let users query for the size limit directly. (But for reference, we're talking about size limits around 100MB, so ... just don't ask for gobs of data.)

Reply:
👍  good with leaving it, simple works and being draconian with rules seems to be the standard so I think it does make sense to not have this fuzzy functionality where things may or may error depending on the underlying logs

Reply:
Agreed. Thanks!